### ISSUE TITLE
EF Code Generation Error

### ISSUE THREAD

**Issue by dcastenholz** [msawczyn/EFDesigner#78]
Putting a Unidirectional Association and a Bidirectional Association on the same 2 entities creates a runtime error for Entity Framework (i.e. One or more validation errors were detected during model generation: PressRelease_Id: Name: Each property name in a type must be unique. Property name 'PressRelease_Id' is already defined.)  The PressRelease_Id property is mapped to the same table twice.  (The code shows the mapping in the 2 individual entities, but they end up on the same table.)

Obviously, `public static string CreateForeignKeyColumnSegmentEF6(Association association, List<string> foreignKeyColumns)` is designed to force them to be unique, but because `public void WriteDbContextEF6(ModelRoot modelRoot)` contains `foreignKeyColumns.Clear();`, and because the 2 conflicting properties are in different passes through the code, the Id is not incremented to make it unique.

One thought would be to change `foreignKeyColumns` to include the name of the table that will actually have the key added to it (i.e. "MyTable:OtherTable_Id").  This might require handling the unidirectional associations differently than the bidirectional ones.  If it was done that way, the `foreignKeyColumns.Clear();` could be removed.  Another idea is to disallow this in the designer.  Probably both would be best.

Here is an example of the generated code with the error:

   ```
modelBuilder.Entity<global::PressReleaseService.EntityFramework.Model.PressRelease>()
			            .HasMany(x => x.PressReleaseDetails)
			            .WithRequired()
			            .Map(x => x.MapKey("PressRelease_Id"));
...
modelBuilder.Entity<global::PressReleaseService.EntityFramework.Model.PressReleaseDetail>()
			            .HasRequired(x => x.PressRelease)
			            .WithMany(x => x.PressReleaseDetailHistory)
			            .Map(x => x.MapKey("PressRelease_Id"));
```

Even though these code segments are associated with different modelBuilder.Entity.ToTable statements, they map a property to the same table.

The obvious workaround is to rename the Bidirectional Association property in the designer.

Thanks for a GREAT product.
Daniel

**Comment by msawczyn** [comment]
This is great stuff - thanks! I'll put it on the list of things to address.

I'm focused on working the `IdentityDbContext` feature right now ... this is the next thing to handle after that.

**Comment by msawczyn** [comment]
Fixed in v1.3.0.4

**Comment by msawczyn** [comment]
Closing since fix was released.
